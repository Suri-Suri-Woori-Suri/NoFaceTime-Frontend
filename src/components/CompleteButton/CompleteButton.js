import React from 'react';
import styles from './CompleteButton.module.css';

const CompleteButton = ({
  currentUser,
  setCurrentUser,
  buttonName,
  fetchToServer,
  setExistMember,
  closeModal
}) => {

  const fetch = async () => {
    const data = await fetchToServer();
    console.log(data);

    if ('rooms' in data) setCurrentUser({ ...currentUser, rooms: data.rooms });
    if ('groups' in data) setCurrentUser({ ...currentUser, groups: data.groups });
    if ('updatedMembers' in data) setExistMember([...data.updatedMembers]);

    closeModal();
  };

  return (
    <button
      className={styles.CompleteButton}
      onClick={fetch}>
      {buttonName}
    </button>
  );
};

export default CompleteButton;
