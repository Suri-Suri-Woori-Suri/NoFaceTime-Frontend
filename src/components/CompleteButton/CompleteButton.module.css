.CompleteButton {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100px;
  height: 40px;
  background: #274ba0;
  color: #ffffff;
  border-color: #274ba0;
  border-width: 1px;
  border-style: solid;
  border-radius: 20px 20px 20px 20px;
  font-family: "Oswald";
  font-weight: 500;
  font-size: 1rem;
  text-align: center;
}

.CompleteButton:hover {
  transform: translateY(2px);
}
