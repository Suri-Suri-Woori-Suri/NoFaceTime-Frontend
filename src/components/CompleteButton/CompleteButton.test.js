import React from 'react';
import { mount } from 'enzyme';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import CompleteButton from './CompleteButton';

describe('<CompleteButton />', () => {
  afterEach(cleanup);

  const mockButtonName = '완료';
  const mockOnClick = jest.fn();

  const wrapper = mount(
    <CompleteButton
      buttonName={mockButtonName}
      onClick={mockOnClick} />
  );

  it('should be contain button', () => {
    const button = wrapper.findWhere(node =>
      node.type() === 'button');

    expect(button.text()).toBe(mockButtonName);
  });

  it('should be render button name correctly', () => {
    expect(wrapper.containsMatchingElement(mockButtonName)).toEqual(true);
  });

  it('should be invoked when button is clicked', () => {
    render(
      <CompleteButton
        buttonName={mockButtonName}
        onClick={mockOnClick} />
    );

    fireEvent.click(screen.getByText(mockButtonName));
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
