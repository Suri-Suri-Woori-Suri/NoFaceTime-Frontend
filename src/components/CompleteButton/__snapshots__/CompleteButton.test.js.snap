// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompleteButton /> matches snapshot 1`] = `
<CompleteButton
  buttonName="완료"
  onClick={[MockFunction]}
>
  <button
    className="CompleteButton"
    onClick={[MockFunction]}
  >
    완료
  </button>
</CompleteButton>
`;
